woocommerce/woocommerce-android
# Issue/4640 Forced firmware update with the SDK 2.0

Forced firmware update with the SDK 2.0

Closes: #4640

### Description
The PR contains refactoring and the implementation of the UI for forced fw updates.

Keep in mind:
* Optional update doesn't work
* Detailed screen doesn't work too
* Cancelation of the forced update is not implemented (another issue)

### Testing instructions
* Open order eligible for payment
* Try collect payment
* Notice that sometimes (currently simulator behaves randomly for testing perposes) forced update happens

### Images/gif

https://user-images.githubusercontent.com/4923871/130791135-588ab81e-b7b9-4039-9e1f-d2145d871905.mp4
